### PR TITLE
Add Update subscriptions pools API call

### PR DESCRIPTION
**Pull request by tumido**
Adding a call to Candlepin API
`HTTP PUT /owners/<owner_key>/subscriptions`
with optional parameters `auto_create_owner` and `lazy_regen`

Cite from API documentation [[ref]](http://www.candlepinproject.org/swagger/?url=candlepin/swagger-2.0.25.json#!/owners/refreshPools):
_Tickle an owner to have all of their entitlement pools synced with their subscriptions._

We're using this call in our app (Red Hat Ethel) and since we use `python-rhsm` for every other request to Candlepin, it would be nice to have this one also covered in there (`python-rhsm` can track future API changes better than we can)

### CODE REVIEW

File: python-rhsm/src/rhsm/connection.py [hunk 1]
```diff
@@ -1405,6 +1405,20 @@ def getSubscriptionList(self, owner_key):
         results = self.conn.request_get(method)
         return results
 
+    def updateSubscriptionList(self, owner_key, auto_create_owner=False, lazy_regen=True):
```
Comment (kahowell) [inline python-rhsm/src/rhsm/connection.py]: Let's not specify the defaults for the parameters `auto_create_owner` and `lazy_regen`. Instead, in python-rhsm, let the params default to `None`... This way the defaults are specified in only one place (server-side, in candlepin code).

File: python-rhsm/src/rhsm/connection.py [hunk 2]
```diff
@@ -1405,6 +1405,20 @@ def getSubscriptionList(self, owner_key):
...
+        """
+        Update subscriptions for a particular owner.
+        """
+        method = "/owners/%s/subscriptions" % self.sanitize(owner_key)
```
Comment (kahowell) [inline python-rhsm/src/rhsm/connection.py]: Missing `?` for the start of query params...

Comment (kahowell) [inline python-rhsm/src/rhsm/connection.py]: By this I mean, in this method, `?` is never added to the URL, but it should be if params are added.

File: python-rhsm/src/rhsm/connection.py [hunk 3]
```diff
@@ -1405,6 +1405,20 @@ def getSubscriptionList(self, owner_key):
...
+
+        if auto_create_owner:
+            method += "&auto_create_owner=true"
+        if lazy_regen:
```
Comment (kahowell) [inline python-rhsm/src/rhsm/connection.py]: These should be done a little differently. As written if you pass `lazy_regen=False`, then the query param is not added at all, and thus candlepin uses default value which is `true`, which is a little surprising. I would suggest we do something like this instead:

```python
if auto_create_owner is not None:
    method += "&auto_create_owner=%s" % bool(auto_create_owner).lower()
if lazy_regen is not None:
    method += "&lazy_regen=%s" % bool(lazy_regen).lower()
```

File: python-rhsm/src/rhsm/connection.py (hunk 1 above)
Comment (tumido) [inline python-rhsm/src/rhsm/connection.py]: Hm, I've chosen this approach, so it's consitant with other methods like `regenEntitlementCertificates` or `getEntitlementList`, etc. I've used Cadnlepin defaults as values here, but it make sense to leave it None as well. I'll change it

File: python-rhsm/src/rhsm/connection.py (hunk 2 above)
Comment (tumido) [inline python-rhsm/src/rhsm/connection.py]: Sure thing, lol.

**Comment by kahowell** [comment]
ok to test

**Comment by kahowell** [comment]
@tumido looks good! Can you please squash this one to one commit as well?

**Comment by tumido** [comment]
Can you please point me what's wrong? Based on the test report, I can't see anything related to this PR.

**Comment by kahowell** [comment]
@tumido, you can reproduce test failures locally with `./setup.py -q nosetests -c playpen/noserc.ci --randomly-seed=1492542919`, but they're not related to your change... the failures happen in master too. I'll handle those outside of this PR. I'll have the tests re-run and with any luck, we won't see them with a new random seed. I'll merge your PR either way :smile:

bugbot, retest this please

**Comment by kahowell** [comment]
retest this please

**Pull request merged**